### ISSUE TITLE
v3 release

### ISSUE THREAD

**Issue by AidasK** [flowjs/flow.js#353]
@drzraf is v3 stable enough to be released? Maybe with tag v3-alpha1?

**Comment by drzraf** [comment]
I'm not yet using it fully in -production and while working on a project depending heavily on it, I'm (still) finding a couple of problems like the ones I tried to resolve in the recent PR.
I'd suggest waiting a bit more (at least a couple of week after #348 and #350 get merged so I can safely close #346)
Ideally, #325 / #322 would be nice too.

Also it'd be nice to have more reviews, in particular an external global look about prototype/instantiation and functions' arguments for general developer experience.

**Comment by AidasK** [comment]
You have made quite a bit of changes so it’s not that easy to review it and I don’t have enough time to fully commit to it. You are already familiar with entire codebase of flow.js, so, I think, you are a perfect developer to lead v3 version of flow.js. It does not have to be perfect, but if it helps to solve your problem others might join as well. 
Anyway, I will keep my eye on your pull requests, but I won’t be able to suggest some ground breaking changes as it would require more development power. 
My suggestion would be to merge early. Tailor this package to fit your production needs and release an alpha version so that others can join you. Other issues can be solved later.

**Comment by ilessiivi** [comment]
I would like to discuss and decide (merge or close) the PR #354 before cutting a v3 pre-release, I feel like merging it after an alpha release or two would be too major of a change.

I am fairly confident though, that whichever way that PR goes, the project is in a good shape for a new release. I have tested the current v3 branch myself in a production environment too, and I have not heard of any horror stories.

**Comment by bertrandg** [comment]
Hi,
Isn't a time to publish a tag `v3-alpha1`?

I'm using it since few months now and it works well. 👍

**Comment by AidasK** [comment]
@drzraf @evilaliv3 It would be nice to release v3-alpha1. Should we update default branch to v3 or should we merge v3 to master?

**Comment by evilaliv3** [comment]
If you are confident that its ready for production already better to just merge at master (if possible renamed main).

As for @GlobaLeaks and maybe other projects/users we would probably continue to use v2 due the compatibility with other modules (ng-flow)

**Comment by drzraf** [comment]
https://github.com/flowjs/flow.js/pull/354 (https://github.com/flowjs/flow.js/pull/363) broke the testsuite but more importantly cause possible problem regarding `async readFileFn` (see #368) [Any problem there is likely to affect resulting file consistency, so both code and tests are of great importance]


I hope to get back on this in the coming weeks, but I strongly feels it should be resolved with clean code before releasing a v3.
(Ideally get rid of this counter-intuitive recursion/event-handler design (https://github.com/flowjs/flow.js/issues/346#issuecomment-897658743))

**Comment by AidasK** [comment]
Sounds like https://github.com/flowjs/flow.js/pull/368 is a must to merge. So we are waiting for you to finish it and after that we can release a first alpha v3 🤗  

ngx-flow will be updated after a release, so don't worry about that
